packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -652,6 +654,20 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
      * @param [options] Options for the request queue
      */
     async run(requests?: (string | Request | RequestOptions)[], options?: CrawlerAddRequestsOptions): Promise<FinalStatistics> {
+        if (this.running) {
+            throw new Error('This crawler instance is already running, you can add more requests to it via `crawler.addRequests()`.');
+        }
+
+        // executing the run method for the second time explicitly, we need to purge the default RQ
+        if (this.running === false && this.requestQueue?.name === 'default') {
+            await this.requestQueue.drop();
+            this.requestQueue = await RequestQueue.open();
+        }
```

Comment:
I would make this optional. I don't see an immediate use-case, but hardcoding something like this seems dangerous to me.